### ISSUE TITLE
Rag-end2end didn't achieve any improvement in recall score compared to training only with Retriever

### ISSUE THREAD

**Issue by Nipi64310** [arcee-ai/DALM#85]
Thanks a lot for this awesome open-source framework. I used a portion of an open-source reading comprehension dataset for training, with 10,000 training data and 6,685 test data.

Here are the results as shown in the table below, use --top_k 5
|                                                | no prompt | query = prompt + query |
| ---------------------------------------------- | --------- | ---------------------- |
| base-model bge-v1.0-large                      | 0.8192969 | 0.8315632              |
| Retriever with contrastive learning -  epoch 3 | 0.8622288 | 0.8806282              |
| Retriever End2End -       epoch 1              | 0.8674644 | 0.8577412              |
| Retriever End2End -       epoch 2              | 0.8557965 | 0.8560957              |

For the prompt, I used the prompt from the BGE model: 'Represent this sentence for searching relevant passages.' Since the BGE model's mean pooling uses the CLS token features, I made modifications to the mean pooling part of the retrieval module in DALM. During the training process, RAG-End2End used a batch size of 16, while Retriever_Only used a batch size of 40. Other parameters were set to their default values.

The conclusion I have reached is that after fine-tuning with contrastive learning, there is a decent improvement in the top 5 recall score, but rag end2end did not help. I wanted to ask about the hyperparameters, training and testing data used for evaluation in the readme, and if I have missed anything?

RAG-end2end loss:
![image](https://github.com/arcee-ai/DALM/assets/31469418/b100e895-0e65-4cc0-aac9-82d1dc0d08b1)

Retriever with contrastive learning  loss:
![image](https://github.com/arcee-ai/DALM/assets/31469418/c9785de6-1b44-4a6b-9e4a-04d3006c4ad0)

**Comment by shamanez** [comment]
Can you try to do the RAG e2e without PEFT? It would be a fair evaluation
since then we fine-tune all params.

**Comment by Nipi64310** [comment]
Thanks for your response. I switched the retrieval model from bge-large to bge-base and replaced the generation model from the 7b model with the 550m model. I completed the fine-tuning with all the parameters, but unfortunately, there wasn't a significant improvement. Here are the results in the table below.

|                                                | Top 1         | Top 5        |
| ---------------------------------------------- | ------------ | ----------- |
| bge-base                                       | 0.404961984  | 0.636254501 |
| Retriever with contrastive learning -  epoch 3 | 0.467386954  | 0.692276910 |
| Retriever End2End -       epoch 1              | 0.4237655062 | 0.647855142 |

I'm not sure if I should run experiments in the 'experiments /llama-index-10k' directory first to confirm if similar results can be achieved.

**Comment by shamanez** [comment]
Well we observed when using a small generator the improvement is not that much.

**Comment by shamanez** [comment]
Also, batch size can play a significant role in the way we compute the marginal loss in the RAG e2e.
